src/streaming.ts

```diff
@@ -0,0 +1,139 @@
+import { BatchId, Bee, Reference, Signer, Topic, Utils } from '@ethersphere/bee-js'
+import type { SingleOwnerChunk } from '@ethersphere/bee-js/dist/src/chunk/soc'
+import type { ChunkReference } from '@ethersphere/bee-js/dist/src/feed'
+import type { EthAddress } from '@ethersphere/bee-js/dist/src/utils/eth'
+import { FeedType, SwarmFeedHandler } from './feed'
+import {
+  assertBytes,
+  Bytes,
+  bytesToHex,
+  hexToBytes,
+  readUint64BigEndian,
+  serializeBytes,
+  TOPIC_BYTES_LENGTH,
+  TOPIC_HEX_LENGTH,
+  writeUint64BigEndian,
+} from './utils'
+
+export type StreamingFeedData = {
+  timestamp: number
+  reference: ChunkReference
+  updatePeriod: number
+  chunkIndex: number
+}
+
+export interface StreamingFeedChunk<Index = number> extends SingleOwnerChunk {
+  index: Index
+  reference: Reference
+  timestamp: number
+  updatePeriod: number
+}
+
+/** Interface for feed type classes */
+export interface SwarmStreamingFeed<Index> {
+  /** Feed type identifier */
+  readonly type: FeedType
+  /** initialised BeeJS instance */
+  readonly bee: Bee
+  /** get Feed interface with read operations */
+  makeFeedR(
+    topic: Topic | Uint8Array | string,
+    owner: EthAddress | Uint8Array | string,
+    ...options: any[]
+  ): SwarmStreamingFeedR<Index>
+  /** get Feed interface with write and read operations */
+  makeFeedRW(
+    topic: Topic | Uint8Array | string,
+    signer: Signer | Uint8Array | string,
+    options?: any,
+  ): SwarmStreamingFeedRW<Index>
+  /** Get Single Owner Chunk identifier */
+  getIdentifier(topic: Bytes<32>, index: Index): Bytes<32>
+}
+
+/** Swarm Feed Read operations */
+export interface SwarmStreamingFeedR<Index = number> extends SwarmFeedHandler {
+  getIndexForArbitraryTime(lookupTime: number, initialTime?: number, updatePeriod?: number): Promise<Index> | Index
+  getUpdate(initialTime: number, updatePeriod: number, lookupTime?: Index): Promise<StreamingFeedChunk<Index>>
+  getUpdates(initialTime: number, updatePeriod: number): Promise<StreamingFeedChunk<Index>[]>
+}
+
+/** Swarm Feed Read and Write operations */
+export interface SwarmStreamingFeedRW<Index = number> extends SwarmStreamingFeedR {
```

Comment:
you don't use your declared type variable.
```suggestion
export interface SwarmStreamingFeedRW extends SwarmStreamingFeedR<number> {
```

Reply:
also, `setUpdate` is missing.